filament/src/Renderer.cpp

```diff
@@ -716,7 +724,14 @@ FrameGraphId<FrameGraphTexture> FRenderer::colorPass(FrameGraph& fg, const char*
                     data.depth = builder.createTexture("Depth Buffer", {
                             .width = colorBufferDesc.width,
                             .height = colorBufferDesc.height,
-                            .format = TextureFormat::DEPTH24
+                            // If the color attachment requested MS, we assume this means the MS buffer
+                            // must be kept, and for that reason we allocate the depth buffer with MS
+                            // as well. On the other hand, if the color attachment was allocated without
+                            // MS, no need to allocate the depth buffer with MS, if the RT is MS,
+                            // the tile depth buffer will be MS, but it'll be resolved to single
+                            // sample automatically -- which is what we want.
+                            .samples = colorBufferDesc.samples,
```

Comment:
The right hand side of this assignment isn't a true sample count because "0" is a valid value, meaning "auto".

The left hand side is an actual sample count that should always be 1 or greater.